lib/queue.rb

```diff
@@ -1,31 +1,70 @@
 class Queue
-
+  
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(100)
+    @front = @back = -1
   end
-
+  
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 1
+      
+    end
+    
+    if @front == @back
+      raise ArgumentError, "Queue if full!"
+    end
```

Comment:
The problem is here, the way you wrote `dequeue` the front could be equal to the back if the queue is empty as well.  That's why we had set front and back to `-1` when the queue was empty in `initialize`.